### DISCUSSION TITLE
behave fails with exit code 1 even though there is no scenario fails

### DISCUSSION THREAD

**Discussion by fie2lim** [behave/behave#954] [Q&A] [answered]
As part of CI/CD, all features under a directory get triggered for every code merged. Time to time, the overall Behave test fails with exit code 1. It stated that there is 1 feature failed, but 0 scenarios failed. This is very bothersome because the testing job gets marked as failed on the pipeline (which is not true because all scenarios are passing)

I noticed this false negative behavior can happen when new scenarios or features get added, and it gets fixed by itself when another new scenarios/features gets added again. However, I could not figure out what is causing it.  For sure, it's not the CI/CD related issue because I can reproduce the bug when I run locally on my laptop.

Here is the snapshot of the behave summary which states 1 feature failed, but 0 scenario failed (Note that Behave will list the failed scenarios at the bottom summary, but in this case, none is listed because there is no scenario failed).

is this a bug in Behave? Any help will be appreciated. Thank you!

![Screen Shot 2021-09-04 at 10 08 22 AM](https://user-images.githubusercontent.com/82006166/132103525-fb1de811-4928-4734-a20a-0b00730f5c34.png)

**Comment by jenisys** [comment, marked as answer]
Has one of the hooks failed that caused one feature to fail?
Like: `before_feature()`or `after_feature()` hook

**HINT:**
* The `untested` Scenarios and steps are hinting for that.
Probably, a `before_feature()` or a `before_tag()` hook (applied to a feature) failed which caused to skip the scenarios (and their steps) of this feature. Meaning, the failed feature should contain 11 untested scenarios.
* Inspect the console output of the CI test run and look for `HOOK-ERROR`.

**Reply by fie2lim** [reply]
THANK YOU SO MUCH jenisys!!! I followed your instructions to search HOOK-ERROR on the console and found it immediately. Number of untested scenarios matched exactly with the features that have hook-error. Thank you again 😁
